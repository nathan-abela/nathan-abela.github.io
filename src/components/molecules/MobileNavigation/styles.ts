import styled from 'styled-components';

const StyledMobileNavigation = styled.ul`
	display: flex;

	li {
		width: 32px;
		margin-left: 16px;
	}
`;

const Menu = styled.button`
	width: 32px;
	height: 32px;
	padding: 0;
	background: transparent;
	border: 0;
	color: ${({ theme }) => theme.text};
	position: absolute;
	z-index: 100;

	&:hover {
		color: ${({ theme }) => theme.headline};
	}
`;

const Overlay = styled.div`
	position: absolute;
	width: 100%;
	height: 100%;
	top: -50%;
	left: 0;
	background: ${({ theme }) => theme.body};
	display: flex;
	align-items: center;
	justify-content: center;
	opacity: 0;
	pointer-events: none;
	transition: all 0.5s ease;
	z-index: 10;

	&.open {
		opacity: 1;
		pointer-events: auto;
		top: 0;
	}
`;

export { Menu, Overlay, StyledMobileNavigation };
