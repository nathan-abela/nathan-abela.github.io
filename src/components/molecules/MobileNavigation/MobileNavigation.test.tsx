import { render } from '@testing-library/react';

import { MobileNavigation } from './MobileNavigation';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: () => mockDispatch,
}));

describe('Mobile Navigation', () => {
	it('renders mobile navigation correctly', () => {
		const { container } = render(
			// eslint-disable-next-line prettier/prettier
			<MobileNavigation />
		);
		expect(container).toMatchSnapshot();
	});
});
