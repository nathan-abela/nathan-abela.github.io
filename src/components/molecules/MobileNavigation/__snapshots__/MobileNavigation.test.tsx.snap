// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile Navigation renders correctly 1`] = `
<div>
  <ul
    class="styles__StyledMobileNavigation-sc-1619527-0 huyxbx"
  >
    <li>
      <button
        class="styles__StyledThemeToggle-sc-1b7mai8-0 kHwexB"
        title="Switch to light theme"
      >
        <test-file-stub />
      </button>
    </li>
    <li>
      <button
        class="styles__Menu-sc-1619527-1 hKtafq"
      >
        <test-file-stub />
      </button>
    </li>
  </ul>
  <div
    class="styles__Overlay-sc-1619527-2 lnRURg"
  >
    <nav
      class="styles__StyledNavigation-sc-1f4kyre-0"
    >
      <ul
        class="styles__List-sc-1f4kyre-1 ivSflr"
      >
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="/education"
          >
            Education
          </a>
        </li>
        <li>
          <a
            href="/work"
          >
            Work
          </a>
        </li>
        <li>
          <a
            href="/portfolio"
          >
            Portfolio
          </a>
        </li>
        <li
          class="desktop-only"
        >
          <button
            class="styles__StyledThemeToggle-sc-1b7mai8-0 kHwexB"
            title="Switch to light theme"
          >
            <test-file-stub />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
